### PR TITLE
Fix the build script

### PR DESCRIPTION
**Pull request by jaulz**
The recent release 10.0.0-alpha.2 does not point to the correct files in the dist folder because the `package.json` is required.

#### Checklist

- [X] I have read the documentation.
- [X] I have included a pull request description of my changes.
- [ ] I have included some tests.
- [ ] If it's a new feature, I have included documentation updates.

### CODE REVIEW

**Comment by sindresorhus** [comment]
See https://github.com/sindresorhus/got/issues/891, that's not it. It shouldn't even generate in in `dist/source`, but rather `dist`.

**Comment by jaulz** [comment]
In my environment the build result looks like this:
image

This is also how it is defined in `tsconfig.json`. I'm not sure if I understood what you expect.

**Comment by sindresorhus** [comment]
Run `npm install got@10.0.0-alpha.2` and look at `node_modules/got/dist`.

**Comment by jaulz** [comment]
@sindresorhus strange indeed but even the new package does not have any source files

**Comment by sindresorhus** [comment]
`npm install got@10.0.0-alpha.2.2` should be good

**Comment by jaulz** [comment]
@sindresorhus okay, I think I found the issue. Since the `package.json` was required it was also compiled to `dist` and this somehow confused the `npm publish`. I just pushed a commit which reads the `package.json` file and parses it instead.

File: source/index.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
-import packageJson = require('../package.json');
+import { readFileSync } from 'fs';
 import create from './create';
 import {Defaults} from './utils/types.js';
 
+const packageJson = JSON.parse(readFileSync('./../package.json', { encoding: 'utf8' }));
```
Comment (sindresorhus) [inline source/index.ts]: You have to use `__dirname` and path.join here.

Comment (jaulz) [reply source/index.ts]: okay, done 😊

**Comment by sindresorhus** [comment]
I think you need to update https://github.com/sindresorhus/got/blob/76ab7e0a38b0068081c9fbb867ec6be285ca7aef/package.json#L7 too, to `dist/source`

**Comment by sindresorhus** [comment]
You're correct for this PR, but master is different. Note this change: https://github.com/sindresorhus/got/commit/640c30bbe37fa399a6f9c96357d897f45e54f52f#diff-e5e546dd2eb0351f813d63d1b39dbc48R12-R14

**Comment by jaulz** [comment]
Okay, I fixed it now.

**Comment by sindresorhus** [comment]
Thanks for fixing this 🙌

**Pull request merged**